src/Perlang.Tests.Integration/Typing/StringTests.cs

```diff
@@ -41,44 +41,215 @@ public void string_variable_can_be_reassigned()
+                var s2: string = " and this is s2";
+                var s3: string = s1 + s2;
+
+                print(s3);
+                """;
+
+        var output = EvalReturningOutputString(source);
+
+        output.Should()
+            .Be("this is s1 with non-ASCII characters: åäöÅÄÖéèüÜÿŸïÏすし and this is s2");
+    }
+
+    [Fact]
+    public void utf8_string_and_ascii_string_literal_can_be_concatenated()
+    {
+        string source = """
+                var s1: string = "this is s1 with non-ASCII characters: åäöÅÄÖéèüÜÿŸïÏすし";
+                var s3: string = s1 + " and this is s2";
+
+                print(s3);
+                """;
+
+        var output = EvalReturningOutputString(source);
+
+        output.Should()
+            .Be("this is s1 with non-ASCII characters: åäöÅÄÖéèüÜÿŸïÏすし and this is s2");
+    }
+
+    [SkippableFact]
+    public void ascii_string_can_be_concatenated_with_int_and_ascii_string()
+    {
+        string source = """
+                var s1: string = "temperature is ";
+                var i: int = 85;
+                var s2: string = " degrees fahrenheit";
+                var s3: string = s1 + i + s2;
+
+                print(s3);
+                """;
+
+        var output = EvalReturningOutputString(source);
+
+        output.Should()
+            .Be("temperature is 85 degrees fahrenheit");
+    }
+
+    [SkippableFact]
+    public void ascii_string_can_be_concatenated_with_double_and_string()
+    {
+        string source = """
+                var s1: string = "temperature is ";
+                var i: double = 85.2;
+                var s2: string = " degrees fahrenheit";
+                var s3: string = s1 + i + s2;
+
+                print(s3);
+                """;
+
+        var output = EvalReturningOutputString(source);
+
+        output.Should()
+            .Be("temperature is 85.2 degrees fahrenheit");
+    }
+
+    [SkippableFact]
+    public void utf8_string_can_be_concatenated_with_double_and_ascii_string()
+    {
+        string source = """
+                var s1: string = "Den årliga medeltemperaturen i Vasa är ";
+                var d: double = 3.4;
+                var s2: string = " grader Celsius";
+                var s3: string = s1 + d + s2;
+
+                print(s3);
+                """;
+
+        var output = EvalReturningOutputString(source);
+
+        output.Should()
+            .Be("Den årliga medeltemperaturen i Vasa är 3.4 grader Celsius");
+    }
+
+    [SkippableFact]
+    public void utf8_string_can_be_concatenated_with_int()
+    {
+        string source = """
+                var s1: string = "Årets varmaste temperatur (Celsius): ";
+                var i: int = 32;
+                var s2: string = s1 + i;
+
+                print(s2);
+                """;
+
+        var output = EvalReturningOutputString(source);
+
+        output.Should()
+            .Be("Årets varmaste temperatur (Celsius): 32");
+    }
+
```

Comment:
These tests were added while working on this, but they are not yet operational because the underlying (C++) stdlib code isn't there yet. Will be fixed in an upcoming PR.